image-patching/stopgap-script/nuage_overcloud_full_patch.py

```diff
@@ -175,53 +193,40 @@ def install_packages(image):
     cmds_run(['cat <<EOT > nuage_packages \n'
               'yum install %s -y \n'
               'yum install %s -y \n'
-              'EOT' % (NUAGE_DEPENDENCIES, NUAGE_PACKAGES)])
+              'yum install %s -y \n'
+              'EOT' % (NUAGE_DEPENDENCIES, NUAGE_PACKAGES, NUAGE_VRS_PACKAGE)])
     virt_customize_run('nuage_packages -a %s --memsize %s --selinux-relabel' % (image, VIRT_CUSTOMIZE_MEMSIZE))
     cmds_run(['rm -f nuage_packages'])
 
-
-#####
-# Installing VRS and the dependencies for VRS
-#####
-
-def install_vrs(image):
-    cmds_run(['cat <<EOT > vrs_packages \n'
-              'yum install %s -y \n'
-              'EOT' % (NUAGE_VRS_PACKAGE)])
-    virt_customize_run('vrs_packages -a %s --memsize %s --selinux-relabel' % (image, VIRT_CUSTOMIZE_MEMSIZE))
-    cmds_run(['rm -f vrs_packages'])
-
-
 #####
 # Function to create the repo file
 #####
 
-def create_repo_file(reponame, repoUrl, image):
-    cmds_run(['cat <<EOT > create_repo \n'
-              'touch /etc/yum.repos.d/nuage.repo \n'
-              'echo "[Nuage]" >> /etc/yum.repos.d/nuage.repo \n'
-              'echo "name=%s" >> /etc/yum.repos.d/nuage.repo \n'
-              'echo "baseurl=%s" >> /etc/yum.repos.d/nuage.repo \n'
-              'echo "enabled = 1" >> /etc/yum.repos.d/nuage.repo \n'
-              'echo "gpgcheck = 0" >> /etc/yum.repos.d/nuage.repo \n'
-              'EOT' % (reponame, repoUrl)])
+def create_repo_file(reponame, repoUrl, image, publickey, Rpm_Key=False, no_key=False):
+    create_repo = 'cat <<EOT > create_repo \n' \
+                  'touch /etc/yum.repos.d/nuage.repo \n' \
+                  'echo "[Nuage]" >> /etc/yum.repos.d/nuage.repo \n' \
+                  'echo "name=%s" >> /etc/yum.repos.d/nuage.repo \n' \
+                  'echo "baseurl=%s" >> /etc/yum.repos.d/nuage.repo \n' \
+                  'echo "enabled = 1" >> /etc/yum.repos.d/nuage.repo \n' % (reponame, repoUrl)
+    if Rpm_Key:
+        create_repo_cmds = create_repo + 'echo "gpgcheck = 1" >> /etc/yum.repos.d/nuage.repo \n' \
+                                         'echo "gpgkey = /tmp/%s" >> /etc/yum.repos.d/nuage.repo \n' \
```

Comment:
1) Make gpgcheck as argument in the function definition and remove new keys.
2) Make publickey default to ''
3) remove tmp from gpgkey value.